src/thriveopedia/pages/ThriveopediaEvolutionaryTreePage.cs

```diff
@@ -0,0 +1,158 @@
+    }
+
+    public override void UpdateCurrentWorldDetails()
+    {
+        if (CurrentGame == null)
+            return;
+
+        RebuildTree();
+    }
+
+    public override void OnNavigationPanelSizeChanged(bool collapsed)
+    {
+    }
+
+    /// <summary>
+    ///   Clear and then rebuild the evolutionary tree each time we open the page. This way, we ensure the tree is
+    ///   always up to date.
+    /// </summary>
+    public void RebuildTree()
+    {
+        if (!Visible)
+            return;
+
+        // TODO fix the tree for freebuild?
+        if (CurrentGame!.FreeBuild)
+        {
+            evolutionaryTree.Visible = false;
+            disabledInFreebuild.Visible = true;
+            return;
+        }
+
+        // Building the tree relies on the existence of a full history of generations stored in the current game. Since
+        // we only started adding these in 0.6.0, it's impossible to build a tree in older saves.
+        // TODO avoid an ugly try/catch block by actually checking the original save version?
+        try
+        {
+            evolutionaryTree.Clear();
+            speciesHistoryList.Clear();
+
+            evolutionaryTree.Init(CurrentGame.GameWorld.PlayerSpecies);
+            InitFirstGeneration();
+
+            // A possible next step would be to rebuild only when the Thriveopedia as a whole is opened.
+            foreach (var generation in CurrentGame.GameWorld.GenerationHistory)
+            {
+                var record = generation.Value;
+                evolutionaryTree.UpdateEvolutionaryTreeWithRunResults(
+                    record.AutoEvoResult, record.Generation, record.TimeElapsed);
+                speciesHistoryList.Add(record.AllSpecies);
+            }
+        }
+        catch (KeyNotFoundException)
+        {
+            evolutionaryTree.Visible = false;
+            disabledInFreebuild.Visible = true;
+        }
+    }
+
+    private void InitFirstGeneration()
+    {
+        // TODO fix this so it shows player species progression rather than current state
+        speciesHistoryList.Add(new Dictionary<uint, Species>
+        {
+            { CurrentGame!.GameWorld.PlayerSpecies.ID, CurrentGame.GameWorld.PlayerSpecies },
+        });
+    }
+
+    private void UpdateSpeciesPreview(Species species)
+    {
+        speciesPreview.PreviewSpecies = species;
+
+        if (species is MicrobeSpecies microbeSpecies)
+        {
+            hexesPreview.PreviewSpecies = microbeSpecies;
+        }
+        else
+        {
+            GD.PrintErr("Unknown species type to preview: ", species);
+        }
+
+        UpdateSpeciesDetail(species);
+    }
+
+    private void EvolutionaryTreeNodeSelected(int generation, uint id)
+    {
+        UpdateSpeciesPreview(speciesHistoryList[generation][id]);
+    }
+
+    private void UpdateSpeciesDetail(Species species)
+    {
+        speciesDetailsLabel.ExtendedBbcode = TranslationServer.Translate("SPECIES_DETAIL_TEXT").FormatSafe(
```

Comment:
This is also probably shared code that'd be easy to put in the microbe class, right?